### DISCUSSION TITLE
A step back to reconsider downloading 1.86GB model. Is this huge download necessary?

### DISCUSSION THREAD

**Discussion by rajibando** [KoljaB/RealtimeTTS#218] [Q&A] [answered]
I found the script `` in the User's Python Virtual directory. I executed it:
`$ python3 /(Home/UserDir)/RealtimeTTS-master/pyqt6_speed_test.py`

The following error message:
> Traceback (most recent call last):
  File "/(Home/UserDir)/RealtimeTTS-master/pyqt6_speed_test.py", line 1, in <module>
    from RealtimeTTS import (
  File "/(Home/UserDir)/Userpy3VirEnv/lib/python3.11/site-packages/RealtimeTTS/__init__.py", line 1, in <module>
    from .text_to_stream import TextToAudioStream  # noqa: F401
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/(Home/UserDir)/Userpy3VirEnv/lib/python3.11/site-packages/RealtimeTTS/text_to_stream.py", line 2, in <module>
    from .stream_player import StreamPlayer, AudioConfiguration
  File "/(Home/UserDir)/Userpy3VirEnv/lib/python3.11/site-packages/RealtimeTTS/stream_player.py", line 8, in <module>
    import resampy
ModuleNotFoundError: No module named 'resampy

Now what should be my next step regarding the module `resampy`? I have installee `resampy`.

After installing all the dependencies, when I executed:
`$python3 /UserHomeFolder/RealtimeTTS-master/pyqt6_speed_test.py `

`Output`:                                               
Downloading config.json to /UserHomeFolder/models/v2.0.2/config.json...
100%|████████████████████| 4.36k/4.36k [00:00<00:00, 4.48MiB/s]
Downloading model.pth to /UserHomeFolder/models/v2.0.2/model.pth...
  4%|▋                   | 65.6M/1.86G [00:41<20:21, 1.47MiB/s]^Z
[1]+  Stopped

**1.86GB?! Is the program berserk?!**

Again executed: 
`$python3 /UserHomeFolder/RealtimeTTS-master/pyqt6_speed_test.py `
`Output`
Downloading vocab.json to /UserHomeFolder/models/v2.0.2/vocab.json...                                                                                   
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████| 335k/335k [00:00<00:00, 550kiB/s]
Downloading speakers_xtts.pth to /UserHomeFolder/models/v2.0.2/speakers_xtts.pth...                                                                     
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████| 7.75M/7.75M [00:04<00:00, 1.75MiB/s]
Start method has already been set. Details: context has already been set

But 1.86GB is huge! I'm in a dilemma now whether to go ahead.

Please advise.

**Comment by KoljaB** [comment, marked as answer]
Thank you for sharing the details! The 1.86GB download is for the [main Coqui TTS model](https://huggingface.co/coqui/XTTS-v2/blob/v2.0.2/model.pth) (XTTS v2.0.2). For a modern TTS model that is supposed to fully infer locally, this is actually considered small. Many AI models require significant storage space to deliver high-quality results. For example, OpenAI's Whisper large-v2 model for ASR also involves downloading several gigabytes.

These large sizes are due to the neural network weights and configurations needed for the model to function effectively. High-quality TTS models rely on this data to produce natural, accurate speech synthesis. While 1.86GB may seem substantial, it is standard for AI models in this field.

To give you additional context, if you look into capable local language models (LLMs), you'll find that their storage requirements often far exceed TTS or ASR models. Models like GPT-style LLMs frequently require downloads of over 20GB, with some exceeding 100GB just for a single model. This scale is necessary for the level of performance and capability these systems offer.

If you proceed with the download, you’ll gain access to a powerful TTS system capable of producing impressive results. Let me know if you need assistance with anything!

**Reply by rajibando** [reply]
Yes Mr. Beigel, I realised it rather late. After I searched for info across websites. So 1.86GB is to be accepted.
Your taking time to convince a user is really admirable.
Now the problem is that I stopped the download earlier. So how do I begin download of the file again?
Thanks to your support, I know that the engine [coqui/XTTS-v2](https://huggingface.co/coqui/XTTS-v2/blob/v2.0.2/model.pth) is [to be downloaded from here](https://huggingface.co/coqui/XTTS-v2/blob/v2.0.2/model.pth), _but what I am unsure of is_:
(1)  _where to save the file_. Will it be: /UserHomeFolder/models/v2.0.2/ as one single file?
(2) Will it have to be decompressed?
(3) if it has to be decompressed, then how? ...
These are some of the questions that I have.
